File: quan.java/config/src/test/java/quan/config/item/UseEffect.java
package quan.config.item;

import com.alibaba.fastjson.*;
import quan.config.*;

/**
 * 使用效果<br/>
 * 自动生成
 */
public class UseEffect extends Bean {

    protected final int aaa;


    public UseEffect(JSONObject json) {
        super(json);

        this.aaa = json.getIntValue("aaa");
    }

    public final int getAaa() {
        return aaa;
    }


    public static UseEffect create(JSONObject json) {
        String clazz = json.getOrDefault("class", "").toString();
        switch (clazz) {
            case "UseEffect3":
                return UseEffect3.create(json);
            case "UseEffect2":
                return UseEffect2.create(json);
            case "UseEffect":
                return new UseEffect(json);
            default:
                return null;
        }
    }

    @Override
    public String toString() {
        return "UseEffect{" +
                "aaa=" + aaa +
                '}';

    }

}

File: quan.cs/Test/Config/Item/UseEffect.cs
using Newtonsoft.Json.Linq;
using Quan.Config;
using Quan.Utils;

namespace Test.Config.Item
{
    /// <summary>
	/// 使用效果<br/>
	/// 代码自动生成，请勿手动修改
	/// </summary>
    public class UseEffect : Bean
    {
        public readonly int aaa;


        public UseEffect(JObject json) : base(json)
        {
            aaa = json["aaa"]?.Value<int>() ?? default;
        }

        public static UseEffect Create(JObject json)
        {
            var clazz = json["class"].Value<string>() ?? "";
            switch (clazz) 
            {
                case "UseEffect4":
                case "item.UseEffect4":
                    return UseEffect4.Create(json);
                case "UseEffect3":
                case "item.UseEffect3":
                    return UseEffect3.Create(json);
                case "UseEffect2":
                case "item.UseEffect2":
                    return UseEffect2.Create(json);
                case "UseEffect":
                case "item.UseEffect":
                    return new UseEffect(json);
                default:
                    return null;   
            }
        }

        public override string ToString()
        {
            return "UseEffect{" +
                   "aaa=" + aaa.ToString2() +
                   '}';
        }
    }
}